core/controllers/reader.py

```diff
@@ -300,7 +300,130 @@ def get(self, exploration_id):
 
 
 class StorePlaythroughHandler(base.BaseHandler):
-    """Handles a useful playthrough coming in from the frontend to store it."""
+    """Handles a useful playthrough coming in from the frontend to store it. If
+    the playthrough already exists, it is updated in the datastore.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Method that initializes member variables for the handler.
+
+        Attributes:
+            current_exp_issues: ExplorationIssues. The exploration issues domain
+                object.
+            current_issue_schema_version: int. The issue schema version.
+            current_playthrough_id: str|None. The Playthrough ID or None.
+        """
+        super(StorePlaythroughHandler, self).__init__(*args, **kwargs)
+        self.current_exp_issues = None
+        self.current_issue_schema_version = None
+        self.current_playthrough_id = None
+
+    def _find_matching_issue_in_exp_issues(self, playthrough):
+        """Finds an equivalent issue for the given playthrough in the unresolved
+        issues list of the exploration issues model, returns None if it doesn't
+        exist.
+
+        Args:
+            playthrough: Playthrough. The playthrough domain object.
+
+        Returns:
+            int|None. The index at which the issue was found, None otherwise.
+        """
+        for index, issue in enumerate(self.exp_issues.unresolved_issues):
+            if issue.issue_type == playthrough.issue_type:
+                issue_customization_args = issue.issue_customization_args
+                # In case issue_keyname is 'state_names', the ordering of the
+                # list is important i.e. [a,b,c] is different from [b,c,a].
+                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
+                    issue.issue_type]
+                if (issue_customization_args[issue_keyname] ==
+                        playthrough.issue_customization_args[issue_keyname]):
+                    return index
+        return None
+
+    def _move_playthrough_to_correct_issue(self, playthrough, orig_playthrough):
+        """Moves the updated playthrough to its correct issue in the unresolved
+        issues list.
+
+        Args:
+            playthrough: Playthrough. The updated playthrough domain object.
+            orig_playthrough: Playthrough. The original playthrough domain
+                object which is in the now-incorrect issue list.
+        """
+        did_move_playthrough_to_new_issue = True
+        issue_index = self._find_matching_issue_in_exp_issues(playthrough)
+        # Check whether the playthrough can be added to its new issue,
+        # if not, it stays in its old issue.
+        if issue_index is not None:
+            issue = self.current_exp_issues.unresolved_issues[issue_index]
+            if len(issue.playthrough_ids) < (
```

Comment:
Move parentheses to start of condition (`if (len(issue.playthrough_ids) < ...)`), or better yet, combine into one line.

Reply:
Done.